04-ruby-mvc-sinatra/sinatra-rest/readme.md

```diff
@@ -12,24 +12,43 @@
 - Use a global template - layout.erb
 - Use forms to submit POST and GET requests
 
+## Opening Framing
+We've learned about databases using SQL. We've also learned about objects in ruby. But we haven't seen any HTML since Unit 1! What gives? How are we going to display all of the things in our database? How can we create a UI to execute CRUD functionality on our different ruby objects? Sinatra. Today you'll be learning about how to create routes and views. In order to talk about how Sinatra functions, we need to talk about how HTTP requests work. Let's talk about REST.
+
 ## REST
 
-or, REpresentational State Transfer is a convention for making HTTP requests. We use REST to remove
-redundancy in network requests and agree on routes and what they do.
+or, REpresentational State Transfer is a software architectural style(convention) for making HTTP requests. We use REST to remove redundancy in network requests and agree on routes and what they do to create a more maintainable architecture.
+
+Every HTTP request consists of a request **method** and **path**
+
+### Examples of RESTful Routes
+
+Method - The Method basically describes the type of request it will be. They will map to the 4 different CRUD functionalities
 
-Every HTTP request consists of a request **method** and **path** 
+| Method | Crud functionality |
+|---|---|
+|GET| read |
+|POST| create |
+|PUT| update |
+|PATCH| update |
+|DELETE| delete |
 
-### Restful Routes
+Path - The path is essentially where the request is sent(the url)
 
 **Route = Method + Path**
 
 | Method | Path | Usage |
 | --- | --- | --- |
-| GET | `/students/:id?` | Read information about a student or students |
+| GET | `/students` | Read information about all students |
+| GET | `/students/1` | Read information about a student whose id is 1 |
 | POST | `/students` | Creates a new student |
-| PUT | `/students/:id` | Updates existing student with all new content |
-| PATCH | `/students/:id` | Updates existing student with partially new content |
-| DELETE | `/students/:id` | Deletes a student
+| PUT | `/students/1` | Updates existing student with an id of 1 with all new content |
+| PATCH | `/students/1` | Updates existing student with an id of 1 with partially new content |
+| DELETE | `/students/1` | Deletes the student with the id of 1
+
+> note the path(url) doesn't contain any of the words describing the CRUD functionality that needs to execute, thats the methods job. This is that removal of redundancy we were talking about before.
+
+Let's checkout the [espn website](http://espn.go.com/). If we go to a specific players webpage, we can this same sort of structure in the URL. It's a written a little differently but the concept is the same .
```

Comment:
maybe a more rails-y app would be better? ![](http://assets.warpspire.com/images/urldesign/urlparts.1.gif)